File: python/src/verifierloop_analysis/schemas/derived.py
"""DERIVED metrics (mirror of crates/metrics/src/derived.rs).

Schema fixed now; values fill as the loop grows. Never a decision criterion.
"""
from __future__ import annotations

from dataclasses import dataclass
from typing import Optional


@dataclass
class DerivedMetrics:
    # REPORT-ONLY efficiency observation (new coverage / new state per exec).
    efficiency_per_exec: Optional[float] = None
    record_count: int = 0

File: crates/metrics/src/derived.rs
//! DERIVED metrics — schema fixed now, values fill as the loop grows.
//!
//! Computed from CORE across a period. Never a decision criterion on its own.

use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, PartialEq, Default, Serialize, Deserialize)]
pub struct DerivedMetrics {
    /// Secondary efficiency observation: new coverage / new state per exec.
    /// REPORT-ONLY — never a stopping/steering criterion. `None` until computable.
    pub efficiency_per_exec: Option<f64>,
    /// Number of CORE records in the period (a simple running aggregate).
    pub record_count: u64,
    // TODO(derived): richer period aggregates (rates, histograms, trend of states).
}
